internal/controller/api/v1alpha1/agentserviceconfig_types.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AgentServiceConfigSpec defines the desired state of AgentServiceConfig
+type AgentServiceConfigSpec struct {
```

Comment:
How about supporting different platforms? Currently we have BMH but i don't think that all the platforms that we will support need to be enabled, otherwise we may need to install a lot of operators even if the user will use only once type of platform. 
@flaper87 It's relevant for your PR as well